Kudu.Contracts/Deployment/DeploymentInfoBase.cs

```diff
@@ -31,8 +32,13 @@ protected DeploymentInfoBase()
         // Optional.
         // Path of the directory to be deployed to. The path should be relative to the wwwroot directory.
         // Example: "webapps/ROOT"
+        // If a path is provided, this should be set to path passed through OneDeploy sans the file name.
         public string TargetPath { get; set; }
 
+        // Hardcode this value when filename is known i.e. app.jar or app.war
+        // When the filename is unknown parse the file name from the path query parameter.
```

Comment:
Renaming TargetPath -> TargetDirectoryPath and FileName -> TargetFileName would make it clearer.